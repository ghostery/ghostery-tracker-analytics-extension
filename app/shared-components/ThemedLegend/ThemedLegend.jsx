import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';

import './ThemedLegend.scss';

const ThemedLegend = ({ pageEvent }) => {
  const keyIconClasses = type => ClassNames(
    'ThemedLegend__keyIcon', type, { 'ThemedLegend--marginTop': pageEvent },
  );
  const legendInfo = [
    { class: 'green', text: '0-99 ms' },
    { class: 'yellow', text: '100-499 ms' },
    { class: 'red', text: '500+ ms' },
    { class: 'pageEvent', text: 'Page event' },
  ];
  return (
    <div className="ThemedLegend d-flex">
      {legendInfo.map((legendItem) => {
        if (legendItem.class === 'pageEvent' && !pageEvent) { return null; }
        return (
          <div
            key={legendItem.class}
            className="ThemedLegend__keyContainer d-flex align-items-center"
          >
            <div className={keyIconClasses(legendItem.class)} />
            <span className="ThemedLegend__keyText">
              {legendItem.text}
            </span>
          </div>
        );
      })}
    </div>
  );
};

ThemedLegend.propTypes = {
  pageEvent: PropTypes.bool,
};

ThemedLegend.defaultProps = {
  pageEvent: false,
};

export default ThemedLegend;
