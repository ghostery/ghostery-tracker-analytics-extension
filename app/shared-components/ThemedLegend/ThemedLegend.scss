@import '../../utils/scss/color-variables.scss';

.ThemedLegend {
  margin-left: 10px;
}

.ThemedLegend__keyContainer {
  margin: 0 5px;

  div,
  span {
    margin: 0 3px;
  }
}

.ThemedLegend__keyIcon {
  opacity: .7;
  width: 9px;
  height: 9px;

  &.green {
    background-color: $info-green;
  }

  &.yellow {
    background-color: $info-yellow;
  }

  &.red {
    background-color: $info-red;
  }

  &.pageEvent {
    background-color: $ghosty-blue;
    width: 1px;
  }

  &.ThemedLegend--marginTop {
    margin-top: 2px;
  }
}

.ThemedLegend__keyText {
  font-size: 10px;
}
