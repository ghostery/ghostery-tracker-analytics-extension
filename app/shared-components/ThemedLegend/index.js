import ThemedLegend from './ThemedLegend.jsx';

export default ThemedLegend;
