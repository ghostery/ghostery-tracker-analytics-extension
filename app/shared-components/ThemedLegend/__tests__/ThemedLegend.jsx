import React from 'react';
import { shallow } from 'enzyme';
import path from 'path';
import fakeProps from 'react-fake-props';

import ThemedLegend from '../ThemedLegend';

describe('app/shared-components/ThemedLegend/', () => {
  describe('Snapshot tests with shallow', () => {
    test('ThemedLegend component should render', () => {
      const componentPath = path.join(__dirname, '../ThemedLegend.jsx');
      const props = fakeProps(componentPath);

      const component = shallow(<ThemedLegend {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
