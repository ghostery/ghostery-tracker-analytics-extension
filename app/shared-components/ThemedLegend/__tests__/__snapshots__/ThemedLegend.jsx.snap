// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/shared-components/ThemedLegend/ Snapshot tests with shallow ThemedLegend component should render 1`] = `
<div
  className="ThemedLegend d-flex"
>
  <div
    className="ThemedLegend__keyContainer d-flex align-items-center"
    key="green"
  >
    <div
      className="ThemedLegend__keyIcon green"
    />
    <span
      className="ThemedLegend__keyText"
    >
      0-99 ms
    </span>
  </div>
  <div
    className="ThemedLegend__keyContainer d-flex align-items-center"
    key="yellow"
  >
    <div
      className="ThemedLegend__keyIcon yellow"
    />
    <span
      className="ThemedLegend__keyText"
    >
      100-499 ms
    </span>
  </div>
  <div
    className="ThemedLegend__keyContainer d-flex align-items-center"
    key="red"
  >
    <div
      className="ThemedLegend__keyIcon red"
    />
    <span
      className="ThemedLegend__keyText"
    >
      500+ ms
    </span>
  </div>
</div>
`;
